src/test/java/uk/gov/pay/directdebit/payments/services/PaymentServiceTest.java

```diff
@@ -329,19 +278,20 @@ public void submitPaymentToProvider_shouldSubmitAndRegisterPaymentSubmittedEvent
 
         when(mockedPaymentProviderFactory.getCommandServiceFor(mandate.getGatewayAccount().getPaymentProvider())).thenReturn(mockedSandboxService);
         when(mockedSandboxService.collect(mandate, payment)).thenReturn(new PaymentProviderPaymentIdAndChargeDate(sandboxPaymentId, chargeDate));
+        
+        when(mockedGovUkPayEventService.storeEventAndUpdateStateForPayment(capturedPayment.capture(), eq(PAYMENT_SUBMITTED))).thenReturn(mockPayment);
 
         Payment returnedPayment = service.submitPaymentToProvider(payment);
-
-        assertThat(returnedPayment.getAmount(), is(payment.getAmount()));
-        assertThat(returnedPayment.getMandate(), is(payment.getMandate()));
-        assertThat(returnedPayment.getState(), is(SUBMITTED_TO_PROVIDER));
-        assertThat(returnedPayment.getDescription(), is("a description"));
-        assertThat(returnedPayment.getReference(), is("a reference"));
-        assertThat(returnedPayment.getProviderId(), is(Optional.of(sandboxPaymentId)));
-        assertThat(returnedPayment.getChargeDate(), is(Optional.of(chargeDate)));
-
-        Payment expectedPaymentWithStatePending = fromPayment(returnedPayment).withState(SUBMITTED_TO_PROVIDER).build();
-        verify(mockedDirectDebitEventService).registerPaymentSubmittedToProviderEventFor(expectedPaymentWithStatePending);
+        assertThat(returnedPayment, is(mockPayment));
+
+        Payment submittedPayment = capturedPayment.getValue();
+        assertThat(submittedPayment.getAmount(), is(payment.getAmount()));
+        assertThat(submittedPayment.getMandate(), is(payment.getMandate()));
+        assertThat(submittedPayment.getState(), is(CREATED));
```

Comment:
Shouldn’t this be `SUBMITTED_TO_PROVIDER`?

Reply:
I'm kind of intercepting the payment before the state is updated to check that the payment service is doing the correct thing before passing it to have the status updated and calculated (which is mocked out).
I think possibly there is some problems with the responsibilities of the class under test that forces me to have to do this, which can hopefully be tidied up as part of the story to remove the DirectDebitEvent stuff.

Reply:
Could you name it something other than `submittedPayment`? Since it’s the payment _before_ it’s been submitted.

Reply:
it's the payment after it's been submitted but before we've changed the state.

It has the provider id and the charge date.

This can be made better as part of refactoring for the DirectDebitEvent stuff.

Reply:
OK, you’ve persuaded me.